src/common/cockpitpipe.c

```diff
@@ -382,6 +383,15 @@ dispatch_error (gint fd,
   gssize ret = 0;
   gsize len;
 
+  if (priv->err_buffer == NULL)
+    {
+      char buf[512];
+      ret = read (priv->err_fd, buf, 512);
+      if (ret > 0)
+        write (2, buf, ret);
```

Comment:
partial write?  blocking?  EINTR?

i guess you'll need to try a bit harder here....

it might even make sense to keep the buffer around and drain it out into fd 2 as it becomes writable...

or maybe it's possible to rethink if there's another way to do this

Reply:
I'd say blocking is fine here. We don't go to the trouble of avoiding to block when writing log messages anywhere else, no?